### DISCUSSION TITLE
[PROPOSAL] Difficulty Adjust Multiplier Calculation

### DISCUSSION THREAD

**Discussion by WitchOfFrost** [ppy/osu#17461] [Ideas]
Hello everyone!

When I was playing around with the Difficulty Adjust mod today, I've come up with a solution as to how to integrate the mod into the current score multiplier systems. Most parts of the explaination can be found on the [Spreadsheet](https://docs.google.com/spreadsheets/d/1OeudPT39Fh3f3ToAtVUeXEByW0Lydw0u7BTPDiPzKMk/edit?usp=sharing), however here is a short rundown of it:

- Difficulty Adjust would use HR Values as a base. Meaning if the HR values were changed, DA would also be changed.
- Each HR Value would attain a weighting that defines the impact an increase/decrease of the value would have on the score multiplier.
- The DA values would simply adjust based around the HR value. Meaning same values = NoMod, same values as HR = HR multiplier, 2x of the HR values = 2x of the HR score (1.12x)
- When reducing the difficulty of a map, DA would apply a flat rate of Penalty for that in addition to the already applied multiplier reduction, based on the new HR Weights. This penalty would add up to a maximum of 0.5x score when reducing all values, which renders DA useless in favor of using EZ.

As I said, this is a short rundown of my proposal. Check out the spreadsheet for a more detailed explaination and lets get this discussion going!

**Comment by bdach** [comment]
is there some reasoning for any of these seemingly arbitrary weightings, penalties, bonuses?

this feels like reinventing pp but for score which i have very mixed feelings about.

**Reply by WitchOfFrost** [reply]
There is not much thought put into the weighting values of the scores. They are mainly a requirement for a calculation like this to work, but the final values are entirely up for debate. However, I have put them in roughly in the way I feel about the different value's impacting the difficulty of a map, hence HP is rated low while CS is rated highly.

**Reply by WitchOfFrost** [reply]
The bonuses are nothing more than an experiment, something I came up with to maybe encourage more people to use DA/HR, since especially HR is on the current (stable) implementation very underplayed in comparison to DT/HD.

As for penalties, the criteria there was to encourage using EZ instead of DA when adjusting down every single value of a map. Hence when going into negative with every value on DA, it will give you a .5x score penalty for just using DA.
